### ISSUE TITLE
Version 3.64.0 and PHP 8

### ISSUE THREAD

**Issue by arnested** [platformsh/platformsh-cli#1015]
platformsh-cli v 3.64.0 fails with PHP 8:
```
PHP Warning:  require_once(phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/Users/arne/.platformsh/bin/platform" in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115

Warning: require_once(phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/Users/arne/.platformsh/bin/platform" in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
PHP Fatal error:  Uncaught Error: Failed opening required 'phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/usr/local/Cellar/php/8.0.0_1/share/php/pear') in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///Users/arne/.platformsh/bin/platform/vendor/composer/autoload_real.php(62): require()
#1 phar:///Users/arne/.platformsh/bin/platform/vendor/composer/autoload_real.php(52): composerRequiref032e6fcc4c17ed28af9c9a1ece8e5b9('0e6d7bf4a5811bf...', 'phar:///Users/a...')
#2 phar:///Users/arne/.platformsh/bin/platform/vendor/autoload.php(7): ComposerAutoloaderInitf032e6fcc4c17ed28af9c9a1ece8e5b9::getLoader()
#3 phar:///Users/arne/.platformsh/bin/platform/bin/platform(22): require('phar:///Users/a...')
#4 /Users/arne/.platformsh/bin/platform(10): require('phar:///Users/a...')
#5 {main}
  thrown in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115

Fatal error: Uncaught Error: Failed opening required 'phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/usr/local/Cellar/php/8.0.0_1/share/php/pear') in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///Users/arne/.platformsh/bin/platform/vendor/composer/autoload_real.php(62): require()
#1 phar:///Users/arne/.platformsh/bin/platform/vendor/composer/autoload_real.php(52): composerRequiref032e6fcc4c17ed28af9c9a1ece8e5b9('0e6d7bf4a5811bf...', 'phar:///Users/a...')
#2 phar:///Users/arne/.platformsh/bin/platform/vendor/autoload.php(7): ComposerAutoloaderInitf032e6fcc4c17ed28af9c9a1ece8e5b9::getLoader()
#3 phar:///Users/arne/.platformsh/bin/platform/bin/platform(22): require('phar:///Users/a...')
#4 /Users/arne/.platformsh/bin/platform(10): require('phar:///Users/a...')
#5 {main}
  thrown in phar:///Users/arne/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
```

Version 3.63.3 worked with PHP 8.

Skimming through the changes from https://github.com/platformsh/platformsh-cli/compare/v3.63.3...v3.64.0 didn't give me any hints. A guess from me would be something in the build process of the phar file...

This happened both after `platform self:update` on my Macbook and in a Docker container based on [php:8-cli-alpine3.12](https://hub.docker.com/layers/php/library/php/8-cli-alpine3.12/images/sha256-44f1d1d7168ea211c58ccffafc1e5585c9b27aae2d23e04da5b10605340f2b72?context=explore) downloading 3.64.0 from GitHub (and also when using the installer).

**Comment by ajardin** [comment]
I also encounter this problem when trying to install the CLI on PHP 8.0, when it still works on an earlier version.

```
$ docker run -it php:8-cli-alpine sh -c 'curl -sS https://platform.sh/cli/installer | php'

Platform.sh CLI installer

Environment check
  [*] The "json" PHP extension is installed.
  [*] The "phar" PHP extension is installed.
  [!] Warning: Git will be needed.
  [*] The "openssl" PHP extension is installed.
  [*] The "pcre" PHP extension is installed.
  [*] One or both of the "mbstring" or "iconv" PHP extensions is installed.
  [*] The "curl" PHP extension is installed.
  [!] The "pcntl" and "posix" extensions are needed for some commands.
  [*] The "allow_url_fopen" setting is on.
  [*] The "apc.enable_cli" setting is off.

Download
  Finding the latest version... done
  Downloading version 3.64.0... done
  Checking file integrity... done
  Checking that the file is a valid Phar... done

Install
  Making the Phar executable... done
  Moving the Phar to your home directory... done
  Executable location: /root/.platformsh/bin/platform

Running self:install command...


Warning: require_once(phar:///root/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/root/.platformsh/bin/platform" in phar:///root/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115

Fatal error: Uncaught Error: Failed opening required 'phar:///root/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/usr/local/lib/php') in phar:///root/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///root/.platformsh/bin/platform/vendor/composer/autoload_real.php(62): require()
#1 phar:///root/.platformsh/bin/platform/vendor/composer/autoload_real.php(52): composerRequiref032e6fcc4c17ed28af9c9a1ece8e5b9('0e6d7bf4a5811bf...', 'phar:///root/.p...')
#2 phar:///root/.platformsh/bin/platform/vendor/autoload.php(7): ComposerAutoloaderInitf032e6fcc4c17ed28af9c9a1ece8e5b9::getLoader()
#3 phar:///root/.platformsh/bin/platform/bin/platform(22): require('phar:///root/.p...')
#4 /root/.platformsh/bin/platform(10): require('phar:///root/.p...')
#5 {main}
  thrown in phar:///root/.platformsh/bin/platform/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
```

```
$ docker run -it php:7-cli-alpine sh -c 'curl -sS https://platform.sh/cli/installer | php'

Platform.sh CLI installer

Environment check
  [*] The "json" PHP extension is installed.
  [*] The "phar" PHP extension is installed.
  [!] Warning: Git will be needed.
  [*] The "openssl" PHP extension is installed.
  [*] The "pcre" PHP extension is installed.
  [*] One or both of the "mbstring" or "iconv" PHP extensions is installed.
  [*] The "curl" PHP extension is installed.
  [!] The "pcntl" and "posix" extensions are needed for some commands.
  [*] The "allow_url_fopen" setting is on.
  [*] The "apc.enable_cli" setting is off.

Download
  Finding the latest version... done
  Downloading version 3.64.0... done
  Checking file integrity... done
  Checking that the file is a valid Phar... done

Install
  Making the Phar executable... done
  Moving the Phar to your home directory... done
  Executable location: /root/.platformsh/bin/platform

Running self:install command...

Copying resource files... done

Setting up autocompletion... skipped (unsupported shell)

To set up the CLI, add the following lines to your shell configuration file:
# BEGIN SNIPPET: Platform.sh CLI configuration
HOME=${HOME:-'/root'}
export PATH="$HOME/"'.platformsh/bin':"$PATH"
if [ -f "$HOME/"'.platformsh/shell-config.rc' ]; then . "$HOME/"'.platformsh/shell-config.rc'; fi # END SNIPPET
```

**Comment by pjcdawkins** [comment]
This is the suspicious change:
https://github.com/symfony/polyfill-mbstring/compare/v1.17.1...v1.19.0

**Comment by pjcdawkins** [comment]
Yes so it's in the build, which includes `.php` files, not `.php8` files :facepalm:

**Comment by pjcdawkins** [comment]
This should be fixed in [v3.64.1](https://github.com/platformsh/platformsh-cli/releases/tag/v3.64.1) - available on GitHub now, and it'll take around 10 minutes to get into the main installer
